org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/FinalValue.java

```diff
@@ -125,4 +126,10 @@ public boolean setIfAbsent(final Callable<VALUE> producer) {
   public boolean isSet() {
     return m_value.get() != NULL_VALUE;
   }
+
+  public void ifSet(Consumer<? super VALUE> action) {
```

Comment:
The actual implementation reads the value twice, the value could be changed in the meantime. Unfortunately there is no atomic ifSet method on AtomicReference. Therefore I would suggest to cache the value in a variable in order to be consistent between get() and action-execution:

```
    VALUE value = get();
    if (value != NULL_VALUE) {
      action.accept(value);
    }
```